diff -Nur feeds/packages/utils/freecwmp/Config.in feeds/packages/utils/freecwmp-modify/Config.in
--- feeds/packages/utils/freecwmp/Config.in	2014-03-31 12:31:02.903604585 +0000
+++ feeds/packages/utils/freecwmp-modify/Config.in	2014-03-30 14:47:15.981486489 +0000
@@ -1,4 +1,4 @@
-if PACKAGE_freecwmp-curl || PACKAGE_freecwmp-zstream
+if PACKAGE_freecwmp || PACKAGE_freecwmp
 
 config FREECWMP_SCRIPTS_FULL
 	bool "Install all freecwmp scripts"
diff -Nur feeds/packages/utils/freecwmp/Makefile feeds/packages/utils/freecwmp-modify/Makefile
--- feeds/packages/utils/freecwmp/Makefile	2014-03-31 12:31:02.859606879 +0000
+++ feeds/packages/utils/freecwmp-modify/Makefile	2014-03-30 15:39:52.316525551 +0000
@@ -17,44 +17,29 @@
 PKG_SOURCE_VERSION:=c1a9c1883189fec77cba32c46bc26658ca3f8123
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION)-$(PKG_SOURCE_VERSION).tar.gz
 
-PKG_FIXUP:=autoreconf
-
 PKG_CONFIG_DEPENDS:= \
 	CONFIG_FREECWMP_ACS_MULTI \
 	CONFIG_FREECWMP_ACS_HDM \
 	CONFIG_FREECWMP_DEBUG \
 	CONFIG_FREECWMP_DEVEL
 
-PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(BUILD_VARIANT)/$(PKG_NAME)-$(PKG_VERSION)
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)/$(PKG_NAME)-$(PKG_VERSION)
 
 include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
 
-define Package/freecwmp/Default
+define Package/freecwmp
   SECTION:=utils
   CATEGORY:=Utilities
   TITLE:=CWMP client
-  DEPENDS:=+libfreecwmp +libuci +libubox +libubus +libmicroxml +shflags
+  DEPENDS:=+libfreecwmp +libuci +libubox +libubus +libmicroxml +shflags +libcurl
 endef
 
 define Package/freecwmp/description
  A free client implementation of CWMP (TR-069) protocol
 endef
 
-define Package/freecwmp-curl
-  $(call Package/freecwmp/Default)
-  TITLE+= (using libcurl)
-  DEPENDS+= +libcurl
-  VARIANT:=curl
-endef
-
-define Package/freecwmp-zstream
-  $(call Package/freecwmp/Default)
-  TITLE+= (using libzstream)
-  DEPENDS+= +libzstream
-  VARIANT:=zstream
-endef
-
-define Package/freecwmp-zstream/config
+define Package/freecwmp/config
 	source "$(SOURCE)/Config.in"
 endef
 
@@ -65,55 +50,28 @@
 endef
 endif
 
-TARGET_CFLAGS += \
-	-D_GNU_SOURCE
-
-TARGET_LDFLAGS += \
-	-Wl,-rpath-link=$(STAGING_DIR)/usr/lib
-
-CONFIGURE_ARGS += \
-	--with-libfreecwmp-include-path=$(STAGING_DIR)/usr/include \
-	--with-uci-include-path=$(STAGING_DIR)/usr/include \
-	--with-libubox-include-path=$(STAGING_DIR)/usr/include \
-	--with-libubus-include-path=$(STAGING_DIR)/usr/include
-
-ifeq ($(BUILD_VARIANT),zstream)
-CONFIGURE_ARGS += \
-	--enable-http=zstream \
-	--with-zstream-include-path=$(STAGING_DIR)/usr/include
-endif
-
-ifeq ($(BUILD_VARIANT),curl)
-CONFIGURE_ARGS += \
-	--enable-http=curl
-endif
-
 ifeq ($(CONFIG_FREECWMP_ACS_MULTI),y)
-CONFIGURE_ARGS += \
-	--enable-acs=multi
+CMAKE_OPTIONS += -DACS:STRING=multi
 endif
 
 ifeq ($(CONFIG_FREECWMP_ACS_HDM),y)
-CONFIGURE_ARGS += \
-	--enable-acs=hdm
+CMAKE_OPTIONS += -DACS:STRING=hdm
 endif
 
 ifeq ($(CONFIG_FREECWMP_DEBUG),y)
-CONFIGURE_ARGS += \
-	--enable-debug
+CMAKE_OPTIONS += -DDEBUG=ON
 endif
 
 ifeq ($(CONFIG_FREECWMP_DEVEL),y)
-CONFIGURE_ARGS += \
-	--enable-devel
+CMAKE_OPTIONS += -DDEVEL=ON
 endif
 
-define Package/freecwmp-$(BUILD_VARIANT)/conffiles
+define Package/freecwmp/conffiles
 /etc/config/freecwmp
 /usr/share/freecwmp/defaults
 endef
 
-define Package/freecwmp-$(BUILD_VARIANT)/install
+define Package/freecwmp/install
 	$(INSTALL_DIR) $(1)/usr/sbin
 	$(CP) $(PKG_BUILD_DIR)/bin/freecwmpd $(1)/usr/sbin
 	$(INSTALL_DIR) $(1)/etc/config
@@ -129,5 +87,4 @@
 endif
 endef
 
-$(eval $(call BuildPackage,freecwmp-curl))
-$(eval $(call BuildPackage,freecwmp-zstream))
+$(eval $(call BuildPackage,freecwmp))
