From 65090b32a1c248415c91db6313fe1fd2b787afa5 Mon Sep 17 00:00:00 2001
From: "Wenjie.zhang" <wenjie.zhang@spreadtrum.com>
Date: Wed, 15 Jan 2014 14:46:42 +0800
Subject: [PATCH] add dmsetup tool into lvm2 package

---
 feeds/packages/utils/lvm2/Makefile |   21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git feeds/packages/utils/lvm2/Makefile feeds/packages/utils/lvm2/Makefile
index 71fab77..95f071a 100644
--- feeds/packages/utils/lvm2/Makefile
+++ feeds/packages/utils/lvm2/Makefile
@@ -47,6 +47,21 @@ define Package/lvm2/description
  LVM toolset.
 endef
 
+define Package/dmsetup
+  SECTION:=utils
+  CATEGORY:=Utilities
+  SUBMENU:=disc
+  TITLE:=The Linux Kernel Device Mapper userspace tool
+  URL:=http://sourceware.org/dm/
+  DEPENDS:=+libdevmapper
+endef
+
+define Package/dmsetup/description
+ The dmsetup manages logical devices that use the device-mapper driver. Devices
+ are created by loading a table that specifies a target for each sector
+ (512 bytes) in the logical device.
+endef
+
 CONFIGURE_ARGS += --disable-o_direct
 
 define Build/Compile
@@ -79,9 +94,15 @@ define Package/lvm2/install
 	$(INSTALL_BIN) ./files/lvm2.init $(1)/etc/init.d/lvm2
 endef
 
+define Package/dmsetup/install
+	$(INSTALL_DIR) $(1)/sbin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/tools/dmsetup $(1)/sbin
+endef
+
 define Package/lvm2/conffiles
 /etc/lvm/lvm.conf
 endef
 
 $(eval $(call BuildPackage,libdevmapper))
 $(eval $(call BuildPackage,lvm2))
+$(eval $(call BuildPackage,dmsetup))
-- 
1.7.9.5

